data/src/main/java/org/apache/iceberg/data/DeleteFilter.java

```diff
@@ -139,43 +173,122 @@ protected long pos(T record) {
           CloseableIterable.transform(CloseableIterable.concat(deleteRecords), Record::copy),
           deleteSchema.asStruct());
 
-      Predicate<T> isInDeleteSet = record -> deleteSet.contains(projectRow.wrap(asStructLike(record)));
-      isInDeleteSets.add(isInDeleteSet);
+      isDeleted = isDeleted == null ? record -> deleteSet.contains(projectRow.wrap(asStructLike(record))) :
```

Comment:
Initializing the `isDeleted` as a predicate like`t->false`  will simplify this if-else as: 

```java
isDeleted = isDeleted.or(record -> deleteSet.contains(projectRow.wrap(asStructLike(record))));
```

Reply:
Done.

Reply:
I found I wrote in this way first but change to the current way because of the comment from Ryan, and that sounds reasonable to me.   FYI https://github.com/apache/iceberg/pull/2372/commits/bfd0aebafa2065861a8a4952ab1ac3652dc3dc1d#r603700530.

Just reverted back.